src/newrelic/processor.go

```diff
@@ -456,6 +462,54 @@ func (p *Processor) processConnectAttempt(rep ConnectAttempt) {
 		NewHarvest(time.Now(), app.connectReply.EventHarvestConfig.EventConfigs), p.processorHarvestChan)
 }
 
+func processLogEventLimits(app *App) {
+
+	if nil == app {
+		log.Warnf("processLogEventLimits() called with *App == nil")
+		return
+	}
+
+	if nil == app.info {
+		log.Warnf("processLogEventLimits() called with app.info == nil")
+		return
+	}
+
+	if nil == app.connectReply {
+		log.Warnf("processLogEventLimits() called with  app.connectReply == nil")
+		return
+	}
+
+	// need to compare agent limits to limits returned from the collector
+	// and choose the smallest value
+	finalLogLimit := limits.DefaultMaxLogEvents
+	agentLogLimit := app.info.AgentEventLimits.LogEventConfig.Limit
+
+	agentReportPeriod := float64(limits.DefaultReportPeriod)
+
+	collectorLogLimit := app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit
+	collectorReportPeriod := float64(app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.ReportPeriod)
+
+	// convert agent log limit to sampling period used for log events
+	agentLogLimit = int((float64(agentLogLimit) * collectorReportPeriod) / agentReportPeriod)
+
+	// MSF - POSSIBLE REMOVE FROM RELEASE
+	log.Debugf("ConnectApplication: agent_report_period = %f collector_report_period = %f", agentReportPeriod, collectorReportPeriod) // MSF
+	log.Debugf("ConnectApplication: agent_log_limit = %d agentLogLimit = %d", agentLogLimit, collectorLogLimit)                       // MSF
+
+	if agentLogLimit < collectorLogLimit {
+		finalLogLimit = agentLogLimit
+	} else {
+		finalLogLimit = collectorLogLimit
+	}
+
+	// store final log limit in reply object which is used by rest of code to
+	// establish harvest limits
+	app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit = finalLogLimit
+
+	// MSF - POSSIBLE REMOVE FROM RELEASE
+	log.Debugf("ConnectApplication: finalLogLimit = %d", app.connectReply.EventHarvestConfig.EventConfigs.LogEventConfig.Limit)
```

Comment:
May want to remove

```suggestion
```

Reply:
I removed the comment but I believe this could be useful troubleshooting output and only occurs once when the daemon connects.